system/Cookie/BaseCookie.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Cookie;
+
+use CodeIgniter\Cookie\Exceptions\CookieException;
+use Config\Cookie as CookieConfig;
+
+abstract class BaseCookie
+{
+	/**
+	 * Configuration Properties
+	 *
+	 * @var array
+	 */
+	protected $properties;
+
+	/**
+	 * Cookie Prefix
+	 *
+	 * Set a cookie name prefix to avoid collisions.
+	 *
+	 * @var string
+	 */
+	protected $prefix = '';
+
+	/**
+	 * Cookie Path
+	 *
+	 * Typically will be a forward slash.
+	 *
+	 * @var string
+	 */
+	protected $path = '/';
+
+	/**
+	 * Cookie Domain
+	 *
+	 * Set to `.example-domain.com` for site-wide cookies.
+	 *
+	 * @var string
+	 */
+	protected $domain = '';
+
+	/**
+	 * Cookie Secure
+	 *
+	 * Transmit the cookie over a secure HTTPS connection only.
+	 *
+	 * @var boolean
+	 */
+	protected $secure = false;
+
+	/**
+	 * Cookie HTTP Only
+	 *
+	 * Make the cookie accessible only through the HTTP protocol (no JavaScript)
+	 *
+	 * @var boolean
+	 */
+	protected $httponly = false;
+
+	/**
+	 * Cookie SameSite
+	 *
+	 * Setting for cookie SameSite.
+	 *
+	 * Allowed values are: None - Lax - Strict - ''.
+	 *
+	 * Defaults to `Lax` as recommended in this link:
+	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
+	 *
+	 * @var string
+	 */
+	protected $samesite = 'Lax';
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Constructor.
+	 *
+	 * @param CookieConfig $config
+	 * 
+	 * @throws CookieException
+	 */
+	public function __construct(CookieConfig $config)
+	{
+		$this->properties = get_object_vars($config);
+
+		foreach (get_class_vars(get_class($this)) as $key => $value)
+		{
+			if (property_exists($this, $key) && isset($this->properties[$key]))
```

Comment:
```suggestion
			if (isset($this->properties[$key]))
```
I don't think `property_exists` is needed as you are checking already on the child cookie's properties.